pkg/trimaran/targetloadpacking/targetloadpacking.go

```diff
@@ -0,0 +1,295 @@
+	requestsMultiplier           float64
+	// Exported for testing
+	WatcherBaseUrl = "/watcher"
+)
+
+type TargetLoadPacking struct {
+	handle       framework.FrameworkHandle
+	client       http.Client
+	metrics      watcher.WatcherMetrics
+	eventHandler *trimaran.PodAssignEventHandler
+	// For safe access to metrics
+	mu sync.RWMutex
+}
+
+func New(obj runtime.Object, handle framework.FrameworkHandle) (framework.Plugin, error) {
+	args, err := getArgs(obj)
+	if err != nil {
+		return nil, err
+	}
+	hostTargetUtilizationPercent = args.TargetUtilization
+	requestsMilliCores = args.DefaultRequests.Cpu().MilliValue()
+	watcherAddress = args.WatcherAddress
+	requestsMultiplier, _ = strconv.ParseFloat(args.DefaultRequestsMultiplier, 64)
+
+	podAssignEventHandler := trimaran.New()
+	pl := &TargetLoadPacking{
+		handle: handle,
+		client: http.Client{
+			Timeout: httpClientTimeoutSeconds,
+		},
+		eventHandler: podAssignEventHandler,
+	}
+
+	pl.handle.SharedInformerFactory().Core().V1().Pods().Informer().AddEventHandler(
+		cache.FilteringResourceEventHandler{
+			FilterFunc: func(obj interface{}) bool {
+				switch t := obj.(type) {
+				case *v1.Pod:
+					return isAssigned(t)
+				case cache.DeletedFinalStateUnknown:
+					if pod, ok := t.Obj.(*v1.Pod); ok {
+						return isAssigned(pod)
+					}
+					utilruntime.HandleError(fmt.Errorf("unable to convert object %T to *v1.Pod in %T", obj, pl))
+					return false
+				default:
+					utilruntime.HandleError(fmt.Errorf("unable to handle object in %T: %T", pl, obj))
+					return false
+				}
+			},
+			Handler: podAssignEventHandler,
+		},
+	)
+
+	// populate metrics before returning
+	err = pl.updateMetrics()
+	if err != nil {
+		klog.Warningf("unable to populate metrics initially: %v", err)
+	}
+	go func() {
+		metricsUpdaterTicker := time.NewTicker(time.Second * metricsUpdateIntervalSeconds)
+		for range metricsUpdaterTicker.C {
+			err = pl.updateMetrics()
+			if err != nil {
+				klog.Warningf("unable to update metrics: %v", err)
+			}
+		}
+	}()
+
+	return pl, nil
+}
+
+func (pl *TargetLoadPacking) updateMetrics() error {
+	req, err := http.NewRequest(http.MethodGet, watcherAddress+WatcherBaseUrl, nil)
+	if err != nil {
+		klog.Errorf("new watcher request failed: %v", err)
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	//TODO(aqadeer): Add a couple of retries for transient errors
+	resp, err := pl.client.Do(req)
+	if err != nil {
+		klog.Errorf("request to watcher failed: %v", err)
+		// Reset the metrics to avoid stale metrics. Probably use a timestamp for better control
+		pl.mu.Lock()
+		pl.metrics = watcher.WatcherMetrics{}
+		pl.mu.Unlock()
+		return err
+	}
+	defer resp.Body.Close()
+	klog.V(6).Infof("received status code %v from watcher", resp.StatusCode)
+	if resp.StatusCode == http.StatusOK {
+		data := watcher.Data{NodeMetricsMap: make(map[string]watcher.NodeMetrics)}
+		var metrics = watcher.WatcherMetrics{Data: data}
```

Comment:
The following style is more conventional:

```suggestion
		metrics := watcher.WatcherMetrics{Data: data}
```